NumLib/DOF/MeshComponentMap.cpp

```diff
@@ -176,6 +176,36 @@ MeshComponentMap MeshComponentMap::getSubset(
     return MeshComponentMap(subset_dict, 1);
 }
 
+MeshComponentMap MeshComponentMap::getSubset(
+    std::vector<std::size_t> const& component_ids,
+    MeshLib::MeshSubsets const& mesh_subsets) const
+{
+    assert(component_ids.size() <= _num_components);
```

Comment:
assert is controlled by NDEBUG macro. Better to use if (!something) OGS_FATAL("something") for such cases.